pandas/_libs/tslib.pyx

```diff
@@ -1577,7 +1599,15 @@ cpdef convert_str_to_tsobject(object ts, object tz, object unit,
                     return obj
                 else:
                     # Keep the converter same as PyDateTime's
-                    ts = Timestamp(obj.value, tz=obj.tzinfo)
+                    obj = convert_to_tsobject(obj.value, obj.tzinfo,
+                                              None, 0, 0)
+                    dtime = datetime(obj.dts.year, obj.dts.month, obj.dts.day,
```

Comment:
use the existing routine to do this

Reply:
Are you referring to `create_datetime_from_ts`?

Reply:
yes

Reply:
`create_datetime_from_ts` is a compatibility func specifically for `ints_to_pydatetime` (you can tell because it has several unused arguments that are retained to match the signature of `create_timestamp_from_ts`).

Replacing a one-line call to `datetime(...)` with a call to a one-line func that does exactly the same thing (but with less clarity because of the unused args) doesn't achieve much, with the possible exception of function call overhead.

Reply:
I know I wrote it!

yet you are repeating code, I would prefer not to.

> with the possible exception of function call overhead.

these take / return objects so the overhead is the same.

Reply:
Fine.